### PR TITLE
Add `dp run` and `dp test` commands

### PR DESCRIPTION
**Pull request by swtwsk**
Depends on #6 

---
Keep in mind:
- [ ] Documentation updates
- [X] [Changelog](CHANGELOG.md) updates

### CODE REVIEW

File: data_pipelines_cli/cli_commands/dbt.py [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
 import click
 
+from ..cli_constants import profiles_build_path
+from ..cli_utils import echo_info, echo_subinfo, subprocess_run
+from ..config_generation import copy_config_dir_to_build_dir, generate_profiles_yml
 
-def dbt(command: Tuple[str, ...], target: str) -> None:
-    subprocess.run(["dbt", *command, "--profiles-dir", ".", "--target", target])
+
+def dbt(command: Tuple[str, ...], env: str, generate_profiles: bool = True) -> None:
```
Comment (swtwsk) [inline data_pipelines_cli/cli_commands/dbt.py:10]: I've decided to keep `dp dbt` anyways, as one could want to call a `dbt` command different from `run` or `test`. However, if using it got too unintuitive, we can safely remove `dp dbt` support.

File: data_pipelines_cli/cli_utils.py [hunk 2]
```diff
@@ -29,3 +30,11 @@ def get_argument_or_environment_variable(
...
         sys.exit(1)
     return result
+
+
+def subprocess_run(args: List[str]):
+    try:
+        subprocess.run(args, check=True)
+    except subprocess.CalledProcessError:
+        echo_error(f"{args[0]} has run across an error")
+        sys.exit(1)
```
Comment (swtwsk) [inline data_pipelines_cli/cli_utils.py:40]: This is a bit cluttered fix of CLI behaviour. Up to this point `subprocess.run` has been usually called without `check=True` flag, so ignoring non-zero exit codes. With the help of this helper function, non-zero exit codes will be stopping the CLI.

File: data_pipelines_cli/cli_utils.py [hunk 3]
```diff
@@ -29,3 +30,11 @@ def get_argument_or_environment_variable(
...
         sys.exit(1)
     return result
+
+
+def subprocess_run(args: List[str]):
+    try:
+        subprocess.run(args, check=True)
+    except subprocess.CalledProcessError as err:
+        echo_error(f"{args[0]} has exited with non-zero exit code: {err.returncode}")
+        sys.exit(1)
```
Comment (swtwsk) [inline data_pipelines_cli/cli_utils.py:40]: This is a bit cluttered fix of CLI behaviour. Up to this point subprocess.run has been usually called without check=True flag, so ignoring non-zero exit codes. With the help of this helper function, non-zero exit codes will be stopping the CLI.

File: data_pipelines_cli/cli_commands/dbt.py (hunk 1 above)
Comment (p-pekala) [reply data_pipelines_cli/cli_commands/dbt.py:10]: I think now we should skip dbt. Is someone wants to use other command he can always use dbt directly.

Comment (swtwsk) [reply data_pipelines_cli/cli_commands/dbt.py]: `dp dbt ...` removed

File: data_pipelines_cli/cli_commands/compile.py [hunk 4]
```diff
@@ -60,10 +87,10 @@ def _docker_build(docker_args: DockerArgs) -> None:
...
     generate_profiles_yml(env)
     echo_info("Running dbt commands:")
-    dbt(("deps",), env)
-    dbt(("compile",), env)
-    dbt(("docs", "generate"), env)
-    dbt(("source", "freshness"), env)
+    dbt(("deps",), env, generate_profiles=True)
+    dbt(("compile",), env, generate_profiles=False)
+    dbt(("docs", "generate"), env, generate_profiles=False)
+    dbt(("source", "freshness"), env, generate_profiles=False)
```
Comment (p-pekala) [inline data_pipelines_cli/cli_commands/compile.py:93]: I think we should simplify it by calling method generate_profile before dbt method and either pass it's path to dbt or build the path inside.

File: data_pipelines_cli/cli_utils.py (hunk 3 above)
Comment (p-pekala) [reply data_pipelines_cli/cli_utils.py:40]: I'm thinking if we should use dbt python package to invoke it's commands? It is not a place to change but let's think about that.

**Review by p-pekala — APPROVED** [review]

**Pull request merged**